scripts/main.js

```diff
@@ -78,462 +78,459 @@ let keysObj2 = {
+    }, false);
+
+    keyboardLayout.addEventListener("click", (ev) => {
+      document.getElementById("txtInput").focus();
+      document.childNodes[1].style.overflowY = "auto";
+      ev.stopPropagation();
+    }, false);
+
+    window.addEventListener("click", () => {
+      window.location.hash = "txtAreaTitle";
+      keyboardLayout.style.bottom = "-100vh";
+      setTimeout(() => document.childNodes[1].style.overflowY = "hidden", 10);
+      ;
+    });
+  }
+
+  keyboardDisplayer(input, keyboardLayout);
+
+
+  // KEYBOARD BUILDER
+  let keyboardBuilder = (keyboardArr1, switcher1 = "", keyboardArr2 = null, switcher2 = "", defDsply1 = "flex", defDsply2 = "none") => {
+
+    // KEYS CONTAINER BUILDER
+    let keysContainer1 = document.createElement("div");
+    keysContainer1.classList.add("keysContainer1");
+    keysContainer1.style.display = defDsply1;
+    keysContainer1.style.flexFlow = "row wrap";
+    keysContainer1.style.justifyContent = "center";
+    keysContainer1.style.alignItems = "center";
+    keyboardLayout.appendChild(keysContainer1);
+
+    let keysContainer2 = document.createElement("div");
+    if (keyboardArr2 !== null) {
+      keysContainer2.classList.add("keysContainer2");
+      keysContainer2.style.display = defDsply2;
+      keysContainer2.style.flexFlow = "row wrap";
+      keysContainer2.style.justifyContent = "center";
+      keysContainer2.style.alignItems = "center";
+      keyboardLayout.appendChild(keysContainer2);
+    }
+
+    // SPECIAL BUTTONS INJECTOR
+    let btnInjector = (array) => {
+
+      // button objects with special functions
+      const capsLock = {
+        value: "",
+        style: {
+          width: "13%"
+        },
+        class: {
+          0: "fal",
+          1: "fa-arrow-alt-up",
+        },
+        operator: () => {
+          let btn = document.querySelectorAll(".keysContainer1 button");
+          let indicatorBtn = document.querySelector(".fa-arrow-alt-up div");
+
+          for (let i = 0; i < btn.length; i++) {
+
+            if (/[a-zA-Z]/.test(btn[i].innerText) && btn[i].innerText != switcher2) {
+              let isLowerCase = true;
+
+              if (/[a-z]/.test(btn[i].innerHTML)) {
+                btn[i].innerText = btn[i].innerText.toUpperCase();
+                indicatorBtn.style.background = " rgb(16, 230, 16)";
+                isLowerCase = false;
+              }
+
+              if (isLowerCase) {
+                indicatorBtn.style.background = "rgba(218, 218, 218, 0.6)";
+                btn[i].innerText = btn[i].innerText.toLowerCase();
+              }
+            }
+          }
+        }
+      }
+
+      const backSpace = {
+        value: "",
+        style: {
+          width: "13%"
+        },
+        class: {
+          0: "fal",
+          1: "fa-backspace"
+        },
+        operator: () => {
+          console.clear();
+
+          // if caret already at 0 index, do NOTHING!
+          if (input.selectionStart === 0) return;
```

Comment:
If you try to select a range which starts or ends at caret index 0, this line prevents the range from being removed after hitting back space button.

Reply:
See below -- we both approached the problem of how the caret behaves at 0 index a little differently...now I understand, and my thing isn't needed.  Deleted.